frontend/src/app/services/form.service.ts

```diff
@@ -29,8 +29,9 @@ export class FormService {
     );
   }
 
-  renameForm(form: Form): any {
+  renameForm(form: Form): Observable<FormsResponse> {
     const body = {formId: form.formId.toString(), formName: form.formName};
-    return this.http.put<any>(this.formEndpoint, body).pipe(retry(3));
+    return this.http.put<any>(this.formEndpoint, body)
+    .pipe(retry(3), mergeMap(() => this.getForms()));
```

Comment:
Do we want the response to the PUT request to contain the updated form or the entire forms response again?
This is fine for now if we want to just refresh the entire datasource with another GET after the PUT (however, this may cause some issues with the noSQL datastore being eventual consistency -> you may get stale data that doesn't reflect the put you just made, so we may want to just have the PUT request return the new form/campaign).
Also, don't see a need to do this atm (a GET after the PUT), since no one else other than the current advertiser can modify the data.

Reply:
Also, just as another thought, the 204 success code could just be an indication that the rename succeeded (which is currently returned rn on success) compared to another failure meaning that the object was not successfully renamed/updated.

Reply:
Yes, I agree this is a hacky solution, but given the time constraint might work best. Also the reason I pull all the forms here is because it helps create a more readable test. (Stubbing the renameForm before this was useless, since it didn't return anything, but now we should expect an updated list of forms).

Reply:
Okay, this should work for now. I'm just concerned in prod it may cause stale data issues of when you rename a form -> the form name reverts to the old name once the GET request returns the old stale data, but we can fix it later.